iphone/Classes/TiUIClipboardProxy.m

```diff
@@ -284,40 +275,37 @@ - (id)hasData:(id)args
         }
       },
       YES);
-  return NUMBOOL(result);
+  return result;
 }
 
-- (id)hasText:(id)unused
+- (bool)hasText
 {
-  return NUMBOOL([[self pasteboard] hasStrings]);
+  return [[self pasteboard] hasStrings];
 }
 
-- (id)hasColors:(id)unused
+- (bool)hasColors
 {
-  return NUMBOOL([[self pasteboard] hasColors]);
+  return [[self pasteboard] hasColors];
 }
 
-- (id)hasImages:(id)unused
+- (bool)hasImages
 {
-  return NUMBOOL([[self pasteboard] hasImages]);
+  return [[self pasteboard] hasImages];
 }
 
-- (id)hasURLs:(id)unused
+- (bool)hasURLs
 {
-  return NUMBOOL([[self pasteboard] hasURLs]);
+  return [[self pasteboard] hasURLs];
 }
 
-- (void)setItems:(id)args
+- (void)setItems:(NSDictionary<NSString *, id> *)args
 {
-  NSArray *items = [args objectForKey:@"items"];
-  NSDictionary *options = [args objectForKey:@"options"];
-
-  __block NSMutableArray *result = [[[NSMutableArray alloc] init] retain];
-
+  NSArray<NSDictionary<NSString *, id> *> *items = args[@"items"];
+  __block NSMutableArray<NSDictionary<NSString *, id> *> *result = [[[NSMutableArray alloc] init] retain];
```

Comment:
No need to retain it. Otherwise memory leak.
```suggestion
  __block NSMutableArray<NSDictionary<NSString *, id> *> *result = [[NSMutableArray alloc] init];
```

General rule -
1. alloc and retain increases retain count by 1.
2. release and autorelease decreases retain count by 1.
3. Object will get release completely if retain count is 0.